### ISSUE TITLE
Recent dataset uploads duplicate branches in taxa tree

### ISSUE THREAD

**Issue by jacobcook1995** [ImperialCollegeLondon/safedata_validator#16]
Datasets uploaded using the develop branch seem duplicate branches of the taxa tree (i.e. [here Formicidae is duplicated](https://sandbox.zenodo.org/record/1059375#.YvD0dC8w08Q). As far as I can tell previous uploads do not show this duplication (e.g. the same (or very similar) [dataset uploaded in 2019](https://sandbox.zenodo.org/record/361815#.YvDWqC8w08Q). This problem persists with datasets created using my current feature branch [see](https://sandbox.zenodo.org/record/1091031#.YvD6RS8w08Q), so is an outstanding issue.

**Comment by jacobcook1995** [comment]
Think that this is an edge case due to "Formicidae" being recorded twice in the index, both as a parent of lower taxa and explicitly as "Formicidae #1". As far as I can see https://github.com/ImperialCollegeLondon/safedata_validator/blob/629fccb99bda005f92fd037f25f42909322bf4db/safedata_validator/zenodo.py#L575 has no means to handle this edge case

**Comment by jacobcook1995** [comment]
I made a start on extending `taxon_index_to_html` to allow it to remove repeated entries (identical bar worksheet name). However, it's not a particularly straightforward thing to do. I'm also wondering if it would be cleaner to overwrite hierarchy taxa (those with `None` as a worksheet name) at the taxa.py level?

That said, I've also found another duplication case where multiple unknown species are defined to the same genus level. This case can't be handled by changing how the index is generated in taxa.py (as both have non-`None` worksheet names), so we might have to alter the tree level functions regardless.

All in all I think this is best paused until we get an opportunity to discuss it.
